library/threading.c

```diff
@@ -29,6 +29,14 @@
 
 #include "mbedtls/threading.h"
 
+#if !defined(_WIN32) && (defined(__unix__) || \
+    (defined(__APPLE__) && defined(__MACH__)))
+#include <unistd.h>
+#if !defined(_POSIX_VERSION)
+#define MBEDTLS_THREADING_USE_GMTIME
```

Comment:
I did not put it there on purpose because this is just a short-hand for:
```
#if !defined(_WIN32) && (defined(__unix__) || \
    (defined(__APPLE__) && defined(__MACH__)))
#if !defined(_POSIX_VERSION)
```
Instead of a real Mbed TLS option or public macro as would be implied if it was placed in the header.

Reply:
I see your point, but I still think we shouldn't have more than one definition that mean the same thing ( even if it is done automatically).
In addition, notice that the `check_names.sh` script fails, probably because you are defining a macro with `MBEDTLS_` prefix in a .c file.

Reply:
Right, thanks for letting me know about check_names.sh. I will fix that, but I do not know whether is appropriate to expose the define in the header to remove the duplication

Reply:
Sure, I don't have strong feelings on that, I just prefer to keep it simple, with a single definition. I would like to hear other opinions as well

Reply:
I have now fixed the check-names.sh failure. Will wait for other opinions regarding whether to put the macros on the header.

Reply:
I agree your reservations about making a new macro public when it's actually an internal technical detail.

OTOH, duplication is bad, so in order to lessen the risk of the two definitions getting out of sync in the future, we should have least at a clear comment in front of each of them "this needs to be kept in sync with the condition in x509.c in function ..." in threading.c, and symmetrically in x509.c.